### PR TITLE
Updating Tutorials for 0.3.0

### PR DESCRIPTION
**Pull request by dgmccart**
1. Updated printed devices to include Zarr V3 devices
2. Updated tutorials for chunking configuration
3. Updated ways of accessing available data
4. Changed storage device from zarr to tiff in Getting Started
5. Added JSON files that capture new API fields

### CODE REVIEW

**Comment by dgmccart** [comment]
@aliddell There are 2 areas I need help with:

1.  The descriptions of the Zarr directory in chunked.md and multiscale.md need improvement. 
2. I am also not sure the most elegant way of accessing AvailableData given that get_available_data now returns AvailableDataContext. This applies to the framedata.md and livestream_napari.py.

**Comment by dgmccart** [comment]
Removed chunked.md and multiscale.md since we ran into bugs with the Zarr directory.

**Comment by melissawm** [comment]
Just rebased it, should be ok now!

File: docs/get_started.md [hunk 1]
```diff
@@ -137,34 +134,28 @@ config.video[1].camera.settings.shape = (1280, 720)
 ```
 
 Now we'll configure each output, or sink device.
-For both simulated cameras, we'll be writing to Zarr, a format which supports chunked arrays.
-
+For both simulated cameras, we'll be writing to [TIFF](http://bigtiff.org/), a well-known format for storing image data.
+For now, we'll simply specify the output file name. For more information about setting up chunking, check out the tutorial [Chunking Data for Zarr Storage](./tutorials/chunked.md)
```
Comment (aliddell) [inline docs/get_started.md:138]: This reads to me like chunking and Tiff are related. I would just omit this line.

File: docs/tutorials/access_data/framedata.md [hunk 2]
```diff
@@ -37,8 +38,13 @@ config = runtime.set_configuration(config)
 ```
 ## Working with `AvailableData` objects
 
-During Acquisition, the `AvailableData` object is the streaming interface, and this class has a `frames` method which iterates over the `VideoFrame` objects in `AvailableData`. Once we start acquisition, we'll utilize this iterator method to list the frames. To increase the likelihood of `AvailableData` containing data, we'll utilize the time python package to introduce a delay before we create our `AvailableData` object
+During Acquisition, the `AvailableData` object is the streaming interface. By calling `get_available_data`, you create an `AvailableDataContext` object which manages memory by deleting the `AvailableData` object when it is no longer in use to free up memory. We can create an `AvailableData` object by using a `with` statement, and work with the `AvailableData` object while it exists inside of the `with` loop. 
 
+There may not be data available, in which case our `AvailableData` object would be `None`. To increase the likelihood of `AvailableData` containing data, we'll utilize the `time` python package to introduce a delay before we create our `AvailableData` object.
```
Comment (aliddell) [inline docs/tutorials/access_data/framedata.md:43]: `AvailableData` is never `None` anymore. You can find out if there are any frames with `get_frame_count()`, but if there's no data this value just returns 0.

File: docs/tutorials/access_data/framedata.md [hunk 3]
```diff
@@ -37,8 +38,13 @@ config = runtime.set_configuration(config)
 ```
 ## Working with `AvailableData` objects
 
-During Acquisition, the `AvailableData` object is the streaming interface, and this class has a `frames` method which iterates over the `VideoFrame` objects in `AvailableData`. Once we start acquisition, we'll utilize this iterator method to list the frames. To increase the likelihood of `AvailableData` containing data, we'll utilize the time python package to introduce a delay before we create our `AvailableData` object
+During Acquisition, the `AvailableData` object is the streaming interface. By calling `get_available_data`, you create an `AvailableDataContext` object which manages memory by deleting the `AvailableData` object when it is no longer in use to free up memory. We can create an `AvailableData` object by using a `with` statement, and work with the `AvailableData` object while it exists inside of the `with` loop.
```
Comment (aliddell) [inline docs/tutorials/access_data/framedata.md:41]: I wouldn't worry about naming the `AvailableDataContext` object. I would just tell users how to get the data and that the data is invalidated after exiting the `with` block, so they should make a copy if they want to do something nontrivial with it.

File: docs/tutorials/access_data/framedata.md [hunk 4]
```diff
@@ -52,65 +58,36 @@ time.sleep(0.5)
...
-
-else:
-    # delete the available_data variable
-    # if there is no data in the packet to free up RAM
-    del available_data
+with runtime.get_available_data(0) as available_data:
+
+    # NoneType if there is no available data.
+    # We can only grab frames if data is available.
+    if available_data is not None:
```
Comment (aliddell) [inline docs/tutorials/access_data/framedata.md:65]: We don't need this check anymore. Use this one instead:

```suggestion
    if available_data.get_frame_count() > 0:
```

File: docs/tutorials/setup_acquisition/configure.md [hunk 5]
```diff
@@ -58,14 +58,14 @@ config.video[0].camera.settings.exposure_time_us = 5e4  # 50 ms
...
+`StorageProperties` has 2 attributes `external_metadata_json` and `filename` which are strings of the filename or filetree of the output metadata in JSON format and image data in whatever format corresponds to the selected storage device, respectively. `first_frame_id` is an integer ID that corresponds to the first frame of the current acquisition and is typically 0. `pixel_scale_um` is the camera pixel size in microns. `acquisition_dimensions` is a list of instances of the `StorageDimension` class, one for each acquisition dimension with the fastest changing dimension listed first and the append dimension listed last. For more information on using the `StorageDimension` class, check out [Chunking Data for Zarr Storage](../zarr/chunked.md). `enable_multiscale` is a boolean used to specify if the data should be saved as an image pyramid. See the [Multiscale tutorial](../zarr/multiscale.md) for more information.
```
Comment (aliddell) [inline docs/tutorials/setup_acquisition/configure.md:68]:
```suggestion
`StorageProperties` has 2 attributes `external_metadata_json` and `filename` which are strings of the filename or filetree of the output metadata in JSON format and image data in whatever format corresponds to the selected storage device, respectively. `first_frame_id` is an integer ID that corresponds to the first frame of the current acquisition and is typically 0. `pixel_scale_um` is the camera pixel size in microns. `acquisition_dimensions` is a list of `StorageDimension`, one for each acquisition dimension, ordered from fastest changing to slowest changing. For more information on using the `StorageDimension` class, check out [Chunking Data for Zarr Storage](../zarr/chunked.md). `enable_multiscale` is a boolean used to specify if the data should be saved as an image pyramid. See the [Multiscale tutorial](../zarr/multiscale.md) for more information.
```

File: docs/tutorials/setup_acquisition/select.md [hunk 6]
```diff
@@ -31,9 +31,12 @@ Output of the above code is below:
 <DeviceIdentifier Storage "Zarr">
 <DeviceIdentifier Storage "ZarrBlosc1ZstdByteShuffle">
 <DeviceIdentifier Storage "ZarrBlosc1Lz4ByteShuffle">
+<DeviceIdentifier Storage "ZarrV3">
+<DeviceIdentifier Storage "ZarrV3Blosc1ZstdByteShuffle">
+<DeviceIdentifier Storage "ZarrV3Blosc1Lz4ByteShuffle">
 ```
 
-All identified devices will be listed, and in the case of this tutorial, no cameras were connected to the machine, so only simulated cameras were found. Note that discovered storage devices will also print.
+All identified devices will be listed, and in the case of this tutorial, no camera drivers were installed on the machine, so only simulated cameras were found. Note that discovered storage devices will also print.
```
Comment (aliddell) [inline docs/tutorials/setup_acquisition/select.md:39]: All the (Acquire) camera drivers should be installed, but device discovery depends on the vendor's drivers being installed and the cameras being properly connected.

File: docs/tutorials/setup_acquisition/storage.md [hunk 7]
```diff
@@ -30,9 +30,12 @@ The output of that script will be:
 <DeviceIdentifier Storage "Zarr">
 <DeviceIdentifier Storage "ZarrBlosc1ZstdByteShuffle">
 <DeviceIdentifier Storage "ZarrBlosc1Lz4ByteShuffle">
+<DeviceIdentifier Storage "ZarrV3">
+<DeviceIdentifier Storage "ZarrV3Blosc1ZstdByteShuffle">
+<DeviceIdentifier Storage "ZarrV3Blosc1Lz4ByteShuffle">
 ```
 
-`Acquire` supports streaming data to [bigtiff](http://bigtiff.org/) and [Zarr V2](https://zarr.readthedocs.io/en/stable/spec/v2.html).
+`Acquire` supports streaming data to [bigtiff](http://bigtiff.org/), [Zarr V2](https://zarr-specs.readthedocs.io/en/latest/v2/v2.0.html), and [Zarr V3](https://zarr-specs.readthedocs.io/en/latest/specs.html).
```
Comment (aliddell) [inline docs/tutorials/setup_acquisition/storage.md:38]: You might mention that we support [OME](https://ngff.openmicroscopy.org/latest/)-Zarr specifically, for both Zarr V2 and Zarr V3.

File: docs/tutorials/zarr/multiscale.md [hunk 8]
```diff
@@ -84,11 +123,16 @@ group = zarr.open("out.zarr")
...
+print(group["0"])
 ```
 
 The output will be:
 
+```
+<zarr.core.Array '/0' (300, 3, 10, 1080, 1920) uint8>
+```
+
+TO BE UPDATED:
```
Comment (aliddell) [inline docs/tutorials/zarr/multiscale.md:135]: Should this be here?

File: docs/tutorials/zarr/chunked.md [hunk 9]
```diff
@@ -78,13 +155,16 @@ group = zarr.open(config.video[0].storage.settings.filename)
...
 
 The output will be:
 ```
-<zarr.core.Array '/0' (10, 1, 1080, 1920) uint8>
+<zarr.core.Array '/0' (300, 3, 10, 1080, 1920) uint8>
 ```
-As expected, we have only 1 top level directory, corresponding to the single array in the group. We would expect more than 1 array only if we were writing [multiscale data](multiscale.md). The overall array shape is (10, 1, 1080, 1920), corresponding to 10 frames, 1 channel, and a height and width of 1080 and 1920, respectively, per frame.
+As expected, we have only 1 top level directory, corresponding to the single array in the group.
+We would expect more than 1 array only if we were writing [multiscale data](multiscale.md).
+The overall array shape is (10, 1, 1080, 1920), corresponding to 10 frames, 1 channel, and a height and width of 1080
```
Comment (aliddell) [inline docs/tutorials/zarr/chunked.md:167]: Shape is wrong here. The 300 in line 163 is a bug -- it should be 10 (= append dimension divided by internal dimensions, i.e., 300 / 3 / 10). If the bug fix could only make it through review, the shape you'd expect in this line is (10, 3, 10, 1080, 1920) corresponding to
- 10 time points
- 3 channels
- 10 planes
- 1080 rows
- 1920 columns

**Review by aliddell — COMMENTED** [review]
Some changes we can discuss in sync

File: docs/get_started.md (hunk 1 above)
Comment (dgmccart) [reply docs/get_started.md]: Done

File: docs/tutorials/zarr/multiscale.md (hunk 8 above)
Comment (dgmccart) [reply docs/tutorials/zarr/multiscale.md:135]: This tutorial got added back when Melissa rebased. I will delete again now as discussed and we can add it back when the bug is fixed.

File: docs/tutorials/zarr/chunked.md (hunk 9 above)
Comment (dgmccart) [reply docs/tutorials/zarr/chunked.md:167]: Thanks for the explanation!

This tutorial got added back when Melissa rebased. I will delete again now as discussed and we can add it back when the bug is fixed.

**Pull request merged**